### ISSUE TITLE
Command-line tools and networked files

### ISSUE THREAD

**Issue by GeoffMPV** [mbunkus/mkvtoolnix#1759]
This may not be an issue with mkvtoolnix itself, but this is where I encountered it, so I would be very interested in your explanation.
Using the command-line tools like mkvinfo, it appears that I can't specify a networked video file on another computer on the local LAN using the form:

`smb://other-computer-name/sharename/mydir/myvideo.mkv`

Some applications accept this syntax and others don't, e.g. vlc and mpv will play the above video fine, but the mkvtoolnix commands (and also the ffmpeg/ffprobe commands) won't. Basic unix commands won't either.
I'm doing a project and application to replace a Windows HTPC with a Linux one (you very kindly helped me by email a while ago with installing your suite on my system, currently Mint 17.2), and I've hit a brick wall with this.
I was assuming that the smb syntax was equivalent to the Windows UNC syntax:

`\\other-computer-name\sharename\mydir\myvideo.mkv`

but it's evidently not so. On Windows the UNC form can be used wherever a file pathname is required, neither the user nor the application writer need to do anything special for it or even be aware of it, it's all handled by the OS.
I know that this sort of thing can be done by e.g. setting up mount points on the computers etc, but that shouldn't be necessary. That smb pathname identifies the file uniquely on the LAN and that is all that is logically required. I should be able to copy my packaged application to any computer on the LAN and it should work without any special action or preparation, as it does on Windows.
I already have my application working on Linux just for playing the files with mpv which accepts the smb syntax, but I would like mkvtoolnix to make some enhancements.

**Comment by mbunkus** [comment]
Sorry, but this won't happen. On Windows support for UNC paths is built into the basic OS API functions for file access. On Linux (and the BSDs, MacOS etc.) there's no equivalent. The file access functions offered by the C libraries and the kernel itself are only able to access files on a mounted file system, whether that file system is a remote file system like `smbfs` or a local one.

For accessing files via various protocols (SMB, SSH, FTP…) there are several projects that implement a virtual file system (VFS) in user space. The problem is that there is a plethora of such VFS layers, e.g. the [KIO system from KDE](https://en.wikipedia.org/wiki/KIO) or the [GVFS system from GNOME](https://en.wikipedia.org/wiki/GVfs), and I simply don't want to tie MKVToolNix to any particular desktop system.

Another possibility would be to solely implement support for the `smb` protocol via the Samba libraries, libsmbclient. However, adding support for `smb` in the command line tools is only part of the equation. I would also have to add support for it in the GUI, and as the GUI uses Qt's built-in file dialogs I cannot simply add support for another protocol to them. So the user could only use such URLs on the command line. This would not be an acceptable solution to me.

Apart from that it's quite a lot of effort that I'm simply not willing to invest myself.

**Comment by GeoffMPV** [comment]
Moritz,

Once again I thank you very much indeed for such an informative and very 
fast reply.

I didn't expect any action from you of course, if networked files were 
easy to support you would have done it already, and so would many other 
application writers. I was just very interested in an explanation of 
something quite astonishing having come from Windows.

I spent the entire morning and afternoon today trying to discover why 
these network paths often didn't work and why they were so seldom 
mentioned on technical forums, but I found nothing particularly helpful. 
Your concise reply taught me more than hours of Googling - and I'm 
extremely good at Googling if I say so myself!.

The issue I raised hasn't appeared on GitHub yet (I'm not very familiar 
with the system), it could be removed since it's not really an issue 
with mkvtoolnix at all.

Again, many thanks, it's very much apprceiated.

Best regards
Geoff
